extide/gradle/src/org/netbeans/modules/gradle/api/execute/GradleDistributionManager.java

```diff
@@ -308,6 +313,46 @@ public GradleDistribution defaultDistribution() {
         return ret;
     }
 
+    /**
+     * Lists all the {@link GradleDistribution}s available on the Gradle Home
+     * of this distribution manager. It looks for the <code>$GRADLE_HOME/wrapper/dists</code>
+     * directory for already downloaded distributions.
+     * @return the list of available Gradle distributions from the Gradle Home.
+     * @since 2.10
+     */
+    public List<GradleDistribution> availableLocalDistributions() {
+        List<GradleDistribution> ret = new ArrayList<>();
+        File dists = new File(gradleUserHome, "wrapper/dists"); //NOI18N
```

Comment:
I'd avoid `/` (possible troubles on Windows). Given Path is needed later, better could be `Paths.get(gradleUserName, "wrapper", "dists")`.